--- a/hw/core/sysbus-fdt.c.orig	2024-03-08 00:06:11.569430000 -0500
+++ b/hw/core/sysbus-fdt.c	2024-03-08 00:06:26.739715709 -0500
@@ -493,7 +493,7 @@
 #endif
 #ifdef CONFIG_TPM
     TYPE_BINDING(TYPE_TPM_TIS_SYSBUS, add_tpm_tis_fdt_node),
-    TYPE_BINDING(TYPE_TPM_CRB_SYSBUS, no_fdt_node),
+    TYPE_BINDING(TYPE_TPM_CRB_SYSBUS, add_tpm_tis_fdt_node),
 #endif
     TYPE_BINDING(TYPE_RAMFB_DEVICE, no_fdt_node),
     TYPE_BINDING("", NULL), /* last element */
